crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractJsonSchema.java

```diff
@@ -170,8 +176,28 @@ protected T internalFrom(TypeDef definition, String... ignore) {
       }
 
       final Property possiblyRenamedProperty = updated[0] != null ? updated[0] : property;
-      addProperty(possiblyRenamedProperty, builder, internalFrom(possiblyRenamedProperty.getName(),
-        possiblyRenamedProperty.getTypeRef()));
+      final T schema = internalFrom(possiblyRenamedProperty.getName(), possiblyRenamedProperty.getTypeRef());
+      // only check accessors for description if the property itself hasn't already been annotated
+      if (description[0] == null) {
```

Comment:
Wouldn't it be possible to perform that lookup when we look for the `@JsonProperty` annotation so that we loop over accessors only once?

Reply:
Possible: new commit loops over accessor annotations once looking for both `@JsonProperty` and `@JsonPropertyDescription`